File: src/main/java/com/asap/server/domain/TimeBlockUser.java
package com.asap.server.domain;

import lombok.*;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

@Entity
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class TimeBlockUser extends AuditingTimeEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "user_id")
    private UserV2 user;
    @ManyToOne
    @JoinColumn(name = "time_block_id")
    private TimeBlock timeBlock;
}

File: domain-db/src/main/kotlin/com/asap/server/domain/timeblock/TimeBlockUser.kt
package com.asap.server.domain.timeblock

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne

@Entity
class TimeBlockUser(
    id: Long? = null,
    userId: Long,
    timeBlock: TimeBlock
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = id
        protected set

    @Column(nullable = false, name = "user_id")
    var userId: Long = userId

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "time_block_id")
    var timeBlock: TimeBlock = timeBlock;
}